### ISSUE TITLE
Add ":spell-next", also :spell-replace doesnt play well with :spell

### ISSUE THREAD

**Issue by Droggelbecher** [mawww/kakoune#1247]
The usual workflow for many people (especially those coming from VIM) is to enable spell checking, then going through all the "wrong" words, correcting them, and then possibly disable spell-checking. For this work flow two ingredients are yet missing in kakoune:

1. Any way to make it possible to automatically re-enable highlighting of spelling mistakes after a :spell-replace (a trivial binding won't do, as there is user-interaction in between!). Maybe an option to :spell-replace?

2. Something like ":spell-next" that jumps to the next misspelled word after the cursor. I don't know enough about kakoune yet to answer this myself, but maybe there is a way to jump to the next location with a particular highlight?

**Comment by lenormf** [comment]
>Any way to make it possible to automatically re-enable highlighting of spelling mistakes after a :spell-replace (a trivial binding won't do, as there is user-interaction in between!). Maybe an option to :spell-replace?

You should still be able to see the highlighted words after having modified the buffer, with a recent version of the editor.

>Something like ":spell-next" that jumps to the next misspelled word after the cursor. I don't know enough about kakoune yet to answer this myself, but maybe there is a way to jump to the next location with a particular highlight?

I've been contemplating implementing that for a while, I do have a working solution ready, I'll wait on the resolution of #1250 before making a PR.

**Comment by lenormf** [comment]
This is fixed as per aecc0fbb0f0cfcfec749c2ee873f33b8d7ba1822, I made a mistake when writing the ID of the issue.

**Comment by mawww** [comment]
Yep, seems to work fine now.
